rfcs/20221006-frontend_event_tracking.md

```diff
@@ -0,0 +1,469 @@
+
+1. A small popup banner that shows up to users, which informs them that Openverse collects anonymized analytics information to make the service better. There are "dismiss" and "learn more" calls to action. "Learn more" directs users to:
+2. An analytics opt-out checkbox on the `/privacy` page. We also add information describing our analytics collection, and link to the relevant code in our GitHub repository.
+
+### Events
+
+Below I've created a TypeScript type for all the proposed anlytics events, and a simple function for sending events. For each event you'll see the payload alongside the default values which will be passed to _all_ events.
+
+```ts
+// src/utils/events.ts
+
+// This type is only for illustrative purposes and would be imported from elsewhere.
+type MediaType = "image" | "audio" | "all";
+
+/**
+ * A list of all events allowed by our analytics server. For each event
+ * there is a description and a series of questions the event can help
+ * answer about our users and their behavior.
+ */
+type AnalyticsEvents = {
+  /**
+   * Description: A search performed by the user
+   * Questions:
+   *   - How many searches do we serve per day/month/year?
+   *   - What are the most popular searches in Openverse?
+   *   - Which media types are the most popular?
+   *   - Do most users search from the homepage, or internal searchbar?
+   */
+  SUBMIT_SEARCH: {
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The search term */
+    query: string;
+  };
+  /** Description: The user navigates to a page.
+   * Questions:
+   *   - Which content pages are the most popular?
+   *   - Are search result pages commonly shared?
+   *   - If so, which are most popular?
+   *   - What is the most common user path?
+   */
+  VIEW_PAGE: {
+    /** The title of the page being navigated to (window.document.title)  */
+    title: string;
+    /** The path of the previous page */
+    previous: string;
+    /** The path of the new page */
+    next: string;
+  };
+  /**
+   * Description: Whenever the user sets a filter
+   * Questions:
+   *  - Do most users filter their searches?
+   *  - What % of users use filtering?
+   *  - Which filters are most popular? Least popular?
+   *  - Are any filters so commonly applied they should become defaults?
+   */
+  APPLY_FILTER: {
+    /** The name of the filter (not the user-facing element label) */
+    key: string;
+    /** The value of the filter  */
+    value: string | boolean;
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The search term */
+    query: string;
+  };
+  /**
+   * Description: Whenever a user changes the content type
+   * Questions:
+   *   - Which content types are most popular?
+   *   - Is there interest in the non-supported content types?
+   *   - Do users switch content types? Where in their journeys?
+   */
+  CHANGE_CONTENT_TYPE: {
```

Comment:
I think `content` type is more appropriate here because `all` is also included. (`mediaType` in the frontend repository refers specifically to the kinds of media like images or audio, and `searchType` can include the combined view of `All media`)

Reply:
I think I prefer to keep @obulat's suggestion, as it matches the frontend code better. Does that sound okay @panchovm and @krysal?

Reply:
If that is the convention on the FE that is perfect then! Thanks for explaining it.